handlers/vault.go

```diff
@@ -37,6 +39,12 @@ func SetSecret(c echo.Context) error {
 		key = s.Key
 		value = s.Value
 	}
+
```

Comment:
It looks like this has been already put in here however, not:

```go

type addSecret struct {
	Key   string `json:"key"`
	Value string `json:"value"`
}

type updateSecret struct {
	Key   string `json:"key"`
	Value string `json:"newvalue"`
}
```

You don't need to do this. Json accepts multiple values:

```go
type updateSecret struct {
	Key   string `json:"key"`
	Value string `json:"value,newvalue"`
}
```

This way you would only have one struct. :)

Reply:
You did this. I thought the same when I saw that 😄

Reply:
OH Gods. What was I thinking? :D :D :D :D